### PR TITLE
Fix #889 #899

### LINKED ISSUE #889: Support File Renaming to Remove "Invalid" Characters
### Prerequisites

* [x] Did you read FAQ section in [readme.md](https://github.com/Nandaka/PixivUtil2/blob/master/readme.md)?
* [x] Did you test with the latest [releases](https://github.com/Nandaka/PixivUtil2/releases "latest releases") or [commit](https://github.com/Nandaka/PixivUtil2/commits/master "latest commit") ?
* [x] Did you search for existing issues in [Issues](https://github.com/Nandaka/PixivUtil2/issues)?

### Description

Images are currently downloaded with the title of the artwork as part of the file name. For some systems, this means it tries to save files with invalid characters for some file systems or services. For example, OneDrive doesn't support the filename "86475500_p0 - 紫罗兰永恒花园 外传礼服薇～(*_ω\\\*).jpg" but will work with the asterisks removed.

It would be useful to have a configuration setting that contains characters considered "invalid" and that the program can strip from the directory or file name when saving. It doesn't even need to be enabled by default, but just there for this type of use  case.

### Steps to Reproduce

1. Download an image that has a character considered invalid

**Expected behavior:** The file is saved without invalid characters in the name

**Actual behavior:** The file is saved with invalid characters

### Versions

20201212-beta3

Likely others as well.

### PR DESCRIPTION
**Pull request by bluerthanever**
For details please refer to each commit.
Mainly:
1. implemented `customBadChars` and added description in `readme.md`, may not work well with spaces
2. optimized methods to parse fanbox posts and the html template and the styles in it.

### CODE REVIEW

**Comment by bluerthanever** [comment]
for the link and styles, the logic i wrote may be a little complicated? but it should be able to deal with such situations, even though no one would probably set the texts like this, but just in case:
original:
original
parsed:

**Comment by bluerthanever** [comment]
for the link and styles, the logic i wrote may be a little complicated? but it should be able to deal with such situations, even though no one would probably set the texts like this, but just in case:
original:
original
parsed:
parsed

**Comment by bluerthanever** [comment]
hmm.. i think i am gonna change the css styles a little more...

File: PixivModelFanbox.py [hunk 1]
```diff
@@ -255,8 +310,7 @@ def parseBody(self, jsPost):
                 elif block["type"] == "embed":  # Implement #470
                     embedId = block["embedId"]
                     if embedId in jsPost["body"]["embedMap"]:
-                        embedStr = self.getEmbedData(jsPost["body"]["embedMap"][embedId], jsPost)
-                        self.body_text = f"{self.body_text}<br />{embedStr}"
+                        self.body_text += self.getEmbedData(jsPost["body"]["embedMap"][embedId], jsPost)
```
Comment (LizeHelesta) [inline PixivModelFanbox.py:313]: Sequence `embed` block will wrap to one line, when the author link plural posts at once.

Comment (bluerthanever) [reply PixivModelFanbox.py:313]: hmm. how about we just wrap it up with a `p` tag. everything would be wrapped in `p`. i think that would be neat.

Comment (bluerthanever) [reply PixivModelFanbox.py:313]: i will do some tests....

Comment (bluerthanever) [reply PixivModelFanbox.py:313]: hmm, just tested with posting a post.
1. for images and files, during editing, even if i deleted the empty line between 2 continuous files or images or one of each, after publish, there is still a blank `p` between them.
2. not the same with embed, no empty `p`s between or before them.

i think i will just wrap them in `p` tags, if you don't mind?

Comment (bluerthanever) [reply PixivModelFanbox.py]: sorry, earlier reply was wrong. please ignore.
just tested with posting a post.
1. for images, during editing, even if i deleted the empty line between 2 continuous images, after publish, there seem always to be a blank p between and after them.
2. not the same with embeds or files, no empty ps between or before them.

i think i will just wrap them in p tags, if you don't mind?

**Comment by bluerthanever** [comment]
Now it should be like this:
original:
QQ图片20210129161315

parsed:
Screenshot_2021-01-29 よりアオイ - This is another test

**Comment by bluerthanever** [comment]
why all of a sudden all the test scripts are picking one me....? 
i am gonna hit the gym and fix this later. please comment or anything.

**Comment by bluerthanever** [comment]
hmm. this is weird. i will pull a new one...

**Pull request closed without merging**